ui/pages/confirmations/confirmation/alerts/useAddEthereumChainAlerts.ts

```diff
@@ -0,0 +1,61 @@
+import { ApprovalRequest } from '@metamask/approval-controller';
+import { ApprovalType } from '@metamask/controller-utils';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { AlertActionKey } from '../../../../components/app/confirm/info/row/constants';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { getMemoizedUnapprovedConfirmations } from '../../../../selectors';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+const VALIDATED_APPROVAL_TYPES = [
+  ApprovalType.AddEthereumChain,
+  ApprovalType.SwitchEthereumChain,
+];
+
+export function useAddEthereumChainAlerts(
+  pendingConfirmation: ApprovalRequest<{ id: string }>,
+): Alert[] {
+  const pendingConfirmations = useSelector(getMemoizedUnapprovedConfirmations);
+
+  const t = useI18nContext();
+  return useMemo(() => {
```

Comment:
As per our best practice, can we try and minimise the logic and dependencies in the `useMemo`?

In this case our only dependency can be a boolean such as `hasOriginApprovals`?

Reply:
fixed

Reply:
Minor, but afraid I can't see the change?